alacritty_terminal/src/config/test.rs

```diff
@@ -1,22 +0,0 @@
-use crate::config::{Config, DEFAULT_ALACRITTY_CONFIG};
-
-#[test]
-fn parse_config() {
-    let config: Config =
-        ::serde_yaml::from_str(DEFAULT_ALACRITTY_CONFIG).expect("deserialize config");
-
-    // Sanity check that mouse bindings are being parsed
-    assert!(!config.mouse_bindings.is_empty());
-
-    // Sanity check that key bindings are being parsed
-    assert!(!config.key_bindings.is_empty());
-}
-
-#[test]
-fn default_match_empty() {
-    let default = Config::default();
-
-    let empty = serde_yaml::from_str("key: val\n").unwrap();
-
-    assert_eq!(default, empty);
-}
```

Comment:
I agree that testing the default config like this isn't a good idea anymore... However, I'd really like it if we could test it somehow. Ideally we'd test that our `Default` matches `alacritty.yml` when everything is uncommented. IDK how best to do this however.

Thoughts?

Reply:
Not possible to do this with the options commented out. And it's really not a good idea to try and uncomment things while parsing. That's just a bigger mess than it will benefit us.

Reply:
I feel pretty uncomfortable about pointing people towards a template config file that, at this point, is completely separated from the `Config` (and it's `Default`) implementation. We just need to manually ensure that `alacritty.yml` is updated as the `Config` changes?

This is a tricky issue though. I think ideally (no idea how to make this happen), we'd document the config options on the `Config` struct (and sub-structs) directly, and have a template (with comments) derived from that. This may be possible with a fancy compiler extension, but otherwise I don't see how to do this. So I guess I'm fine with this for now, and we can continue to be vigilant about config changes.

Reply:
We've been doing this already. We just had sanity checks for *some* fields, not for all of them. So, I don't any major problem here.

Reply:
Yea, I guess this is a small-ish loss. I'd still be interested in how this could be better though. Not blocking moving forward with this ATM though.